nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java

```diff
@@ -356,6 +360,107 @@ public void testListTables() {
+  }
+
+  @Test
+  public void testRegisterTableTagName() throws NessieConflictException, NessieNotFoundException {
+    List<String> metadataVersionFiles = metadataVersionFiles(TABLE_NAME);
+    Assertions.assertThat(1).isEqualTo(metadataVersionFiles.size());
+    api.createReference().sourceRefName(BRANCH).reference(Tag.of("tag_1", catalog.currentHash())).create();
+
+    Assertions.assertThatThrownBy(
+                    () -> catalog.registerTable(
+                            TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "`@tag_1"),
+                            "file:" + metadataVersionFiles.get(0)))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("You can only mutate tables when using a branch without a hash or timestamp.");
+  }
+
+  @Ignore
+  public void testRegisterTableMoreThanOneBranch() throws NessieConflictException, NessieNotFoundException {
+    // Dependency on PR #5033 for the backend bug fixes in drop table.
+    List<String> metadataVersionFiles = metadataVersionFiles(TABLE_NAME);
+    Assertions.assertThat(1).isEqualTo(metadataVersionFiles.size());
+    catalog.dropTable(TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@main`"), false);
+    catalog.registerTable(
+            TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@main`"),
+            "file:" + metadataVersionFiles.get(0));
+    Table newTable = catalog.loadTable(TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@main`"));
+    Assertions.assertThat(newTable).isNotNull();
+    TableOperations ops = ((HasTableOperations) newTable).operations();
+    String metadataLocation = ((NessieTableOperations) ops).currentMetadataLocation();
+    Assertions.assertThat("file:" + metadataVersionFiles.get(0)).isEqualTo(metadataLocation);
+    Assertions.assertThat(catalog.tableExists(TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@main`"))).isTrue();
+
+    api.createReference().sourceRefName(BRANCH).reference(Tag.of("branch1", catalog.currentHash())).create();
+
+    List<String> metadataVersionFiles1 = metadataVersionFiles(TABLE_NAME);
+    Assertions.assertThat(1).isEqualTo(metadataVersionFiles1.size());
+    catalog.dropTable(TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@branch1`"));
+    catalog.registerTable(
+            TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@branch1`"),
+            "file:" + metadataVersionFiles1.get(0));
+    Table newTable1 = catalog.loadTable(TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@branch1`"));
+    Assertions.assertThat(newTable1).isNotNull();
+    TableOperations ops1 = ((HasTableOperations) newTable1).operations();
+    String metadataLocation1 = ((NessieTableOperations) ops1).currentMetadataLocation();
+    Assertions.assertThat("file:" + metadataVersionFiles1.get(0)).isEqualTo(metadataLocation1);
+    Assertions.assertThat(catalog.tableExists(TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@branch1`"))).isTrue();
+  }
+
   private String getTableBasePath(String tableName) {
```

Comment:
Add a test case with invalid identifiers, non-null metadata path but metadata file not exist in that path, null metadata location.